ci: Clean up logic for `merge -r`

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114295

Rely on built in bash conditionals for doing the if statement rather
than relying on $?

To avoid issues observed in https://github.com/pytorch/pytorch/pull/111008#issuecomment-1821547141

Signed-off-by: Eli Uriegas <eliuriegas@meta.com>

seemethere committed:
ci: Clean up logic for `merge -r`

Rely on built in bash conditionals for doing the if statement rather
than relying on $?

Signed-off-by: Eli Uriegas <eliuriegas@meta.com>

[ghstack-poisoned]

huydhn reviewed (approved):
LGTM!

seemethere commented:
I also don't know if there's a good way to test this without a YOLO

seemethere commented:
@pytorchbot merge

malfet commented on .github/workflows/trymerge.yml:
```diff
@@ -45,8 +45,8 @@ jobs:
         run: |
```
```suggestion
        shell: bash
        run: |
```

seemethere replied:
Added above

malfet reviewed (approved):
LGTM, please also add `shell: bash` to avoid any unexpected changes

seemethere committed:
Update on "ci: Clean up logic for `merge -r`"


Rely on built in bash conditionals for doing the if statement rather
than relying on $?

To avoid issues observed in https://github.com/pytorch/pytorch/pull/111008#issuecomment-1821547141

Signed-off-by: Eli Uriegas <eliuriegasmeta.com>

[ghstack-poisoned]

seemethere commented:
@pytorchbot merge
